main.tf

```diff
@@ -113,12 +115,13 @@ resource "aws_ec2_transit_gateway_route" "this" {
 }
 
 resource "aws_route" "this" {
-  for_each = { for x in local.vpc_route_table_destination_cidr : x.rtb_id => {
+  for_each = { for x in local.vpc_route_table_destination_cidr : "${x.rtb_id}-${x.cidr}" => {
```

Comment:
One problem I found with this approach is that if you're creating the route table in the same Terraform, Terraform will complain about `The “for_each” map including keys derived from resource attributes that
cannot be determined until apply` because the route table id is not known at plan time. I solved this using the index of the array instead of the rtb_id https://github.com/ronaldour/terraform-aws-transit-gateway/blob/refactor/main.tf#L19.

Btw I have this [fork with a refactor](https://github.com/ronaldour/terraform-aws-transit-gateway/tree/refactor?tab=readme-ov-file#managing-transit-gateway-attachments), would love to get some feedback from people.